doc/adr/0015-time-conversion.rst

```diff
@@ -0,0 +1,151 @@
+.. _time_conversion_semantic_change:
+
+ADR 15: Time conversion semantic change
+=======================================
+
+Date: 2022-11-19
+
+Author(s)
+---------
+
+koslambrou <konstantinos.lambrou@iohk.io>
+
+Status
+------
+
+Draft
+
+Context
+-------
+
+Currently, PAB users need to provide the ``SlotConfig`` in the configuration file, which is passed
+through to the ``Contract`` API, which users can use to convert between a ``Slot`` and a ``POSIXTime``.
+However, the current ``SlotConfig`` representation supposes that the slot length is the same for all
+epochs in the Cardano blockchain, which is not the case.
+For example, during the Byron era, the slot length was 20s, while from Shelley era and onwards, the
+slot length is 1s.
+Therefore, the functions from the ``Ledger.TimeSlot`` module in ``plutus-ledger`` do not compute the
+conversion between ``Slot`` and ``POSIXTime`` the right way.
+The current easiest way to compute the time conversions is to query the local Cardano node on the
+consensus layer, which requires the ``ouroboros-consensus`` dependency.
+
+Decision
+--------
+
+* We will deprecate the ``Ledger.TimeSlot.SlotConfig`` type and all functions in the
+  ``Ledger.TimeSlot`` module using the ``SlotConfig``. The only viable functions are the ones that
+  convert between ``Data.Time`` types and plutus types (types related to ``TxInfo``).
+
+* We will copy the ``Ledger.TimeSlot`` module in the emulator (ideally rename it) and keep it as
+  an internal module. Any functions not used by the emulator will be removed.
+
+* We will move the ``Ledger.Params`` module inside the emulator as an internal module and modify the
```

Comment:
Maybe we can just remove `SlotConfig` from `Params`. It doesn't need to be configurable anymore.

Reply:
Developpers may want to test how there contract behave with a given `SlotConfig`?

Reply:
I think it *does* need to be configurable. The PAB simulator actually uses it to define the `delay` for creating new blocks.

Reply:
Ok!